lightning/src/ln/channelmanager.rs

```diff
@@ -1610,6 +1641,32 @@ impl<M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelManager<M, T, K, F
 		self.list_channels_with_filter(|&(_, ref channel)| channel.is_live())
 	}
 
+	/// Gets a list of payments in random order that are currently pending resolution.
+	///
+	/// This can be useful for figuring out whether or not a payment made needs to be retried.
+	/// In general, if it is not listed here, you should consider retrying it with a caveat: if a
+	/// payment is returned as [`RecentPaymentDetails::Fulfilled`], you may retry it, but should
+	/// not.
+	pub fn list_pending_payments(&self) -> Vec<RecentPaymentDetails> {
+		self.pending_outbound_payments.pending_outbound_payments.lock().unwrap().iter()
```

Comment:
I think the `pending_outbound_payments` field in `OutboundPayments` may be worth renaming to avoid `self.pending_outbound_payments.pending_outbound_payments`.

thoughts? cc // @jkczyz @valentinewallace @TheBlueMatt

Reply:
Yeah, I would be supportive of reaming. Perhaps `id_to_pending_payment`, `payment_by_id`, or the like? Open to other alternatives given we may want to be deliberate about where "pending" is used.